### ISSUE TITLE
Fix axe aria-allowed-attr violations

### ISSUE THREAD

**Issue by mcoker** [patternfly/patternfly#4963]
There are a bunch of aria-allowed-attr failures since updating the docs framework

```
346/1031  /extensions/log-viewer/html/with-popover-open
violations: aria-allowed-attr
--
412/1031  /components/modal/html/basic
violations: aria-allowed-attr
--
413/1031  /components/modal/html/with-help-button
violations: aria-allowed-attr
--
414/1031  /components/modal/html/small
violations: aria-allowed-attr
--
415/1031  /components/modal/html/medium
violations: aria-allowed-attr
--
416/1031  /components/modal/html/large
violations: aria-allowed-attr
--
417/1031  /components/modal/html/without-title
violations: aria-allowed-attr
--
418/1031  /components/modal/html/with-description
violations: aria-allowed-attr
--
419/1031  /components/modal/html/custom-title
violations: aria-allowed-attr
--
420/1031  /components/modal/html/modal-box-as-generic-container
violations: aria-allowed-attr
--
421/1031  /components/modal/html/icon
violations: aria-allowed-attr
--
422/1031  /components/modal/html/info-alert
violations: aria-allowed-attr
--
423/1031  /components/modal/html/default-alert
violations: aria-allowed-attr
--
424/1031  /components/modal/html/success-alert
violations: aria-allowed-attr
--
425/1031  /components/modal/html/warning-alert
violations: aria-allowed-attr
--
427/1031  /components/modal/html/danger-alert
violations: aria-allowed-attr
--
508/1031  /components/popover/html/top
violations: aria-allowed-attr
--
509/1031  /components/popover/html/bottom
violations: aria-allowed-attr
--
510/1031  /components/popover/html/right
violations: aria-allowed-attr
--
511/1031  /components/popover/html/left
violations: aria-allowed-attr
--
512/1031  /components/popover/html/left-with-top-and-bottom-positions
violations: aria-allowed-attr
--
513/1031  /components/popover/html/without-headerfooter
violations: aria-allowed-attr
--
514/1031  /components/popover/html/no-padding
violations: aria-allowed-attr
--
515/1031  /components/popover/html/bottom-with-left-and-right-positions
violations: aria-allowed-attr
--
516/1031  /components/popover/html/width-auto
violations: aria-allowed-attr
--
517/1031  /components/popover/html/default-alert-popover
violations: aria-allowed-attr
--
518/1031  /components/popover/html/popover-with-icon-in-the-title
violations: aria-allowed-attr
--
519/1031  /components/popover/html/info-alert-popover
violations: aria-allowed-attr
--
520/1031  /components/popover/html/warning-alert-popover
violations: aria-allowed-attr
--
521/1031  /components/popover/html/success-alert-popover
violations: aria-allowed-attr
--
522/1031  /components/popover/html/danger-alert-popover
violations: aria-allowed-attr
--
906/1031  /components/card/html-demos/status-card-expanded-with-popover
violations: aria-allowed-attr
--
955/1031  /components/modal/html-demos/basic
violations: aria-allowed-attr
--
956/1031  /components/modal/html-demos/scrollable-content
violations: aria-allowed-attr
--
957/1031  /components/modal/html-demos/medium
violations: aria-allowed-attr
--
958/1031  /components/modal/html-demos/top-aligned
violations: aria-allowed-attr
--
959/1031  /components/modal/html-demos/large
violations: aria-allowed-attr
--
960/1031  /components/modal/html-demos/modal-with-form
violations: aria-allowed-attr
--
1003/1031 /components/table/html-demos/column-management-modal
violations: aria-allowed-attr
--
1014/1031 /components/tabs/html-demos/modal-tabs
violations: aria-allowed-attr
--
1024/1031 /components/wizard/html-demos/basic
violations: aria-allowed-attr
--
1025/1031 /components/wizard/html-demos/nav-expanded-mobile
violations: aria-allowed-attr
--
1026/1031 /components/wizard/html-demos/with-drawer-closed
violations: aria-allowed-attr
--
1027/1031 /components/wizard/html-demos/with-drawer-expanded
violations: aria-allowed-attr
--
1028/1031 /components/wizard/html-demos/with-drawer-and-informational-step
violations: aria-allowed-attr
```

**Comment by mcoker** [comment]
@jessiehuff this is a result of removing `role="dialog"` with elements that have `aria-modal="true"` in the core examples, which was done to fix a VO issue described in https://github.com/patternfly/patternfly/issues/2538

WDYT we could do about fixing this? One solution could be to update the modal/popover examples to be full screen. Another could be to also remove `aria-modal="true"` for in-page examples and make a note that it needs to be added, like `role="dialog"` is currently - https://github.com/patternfly/patternfly/blob/main/src/patternfly/components/ModalBox/examples/ModalBox.md#accessibility

> **Note:** `role="dialog"` is not included on the static examples on this page as it interferes with VoiceOver. Refer to the react examples to see the role in use. **Required**
